modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/processors/ExternalRefProcessor.java

```diff
@@ -83,64 +84,88 @@ public String processRefToExternalSchema(String $ref, RefFormat refFormat) {
 
             String file = $ref.split("#/")[0];
             if (schema.get$ref() != null) {
-                RefFormat format = computeRefFormat(schema.get$ref());
-                if (isAnExternalRefFormat(format)) {
-                    schema.set$ref(processRefToExternalSchema(schema.get$ref(), format));
-                } else {
-                    processRefToExternalSchema(file + schema.get$ref(), RefFormat.RELATIVE);
+                processRefSchema(schema,file);
+            }
+
+            if(schema instanceof ComposedSchema){
+                ComposedSchema composedSchema = (ComposedSchema) schema;
+                if (composedSchema.getAllOf() != null){
+                    for(Schema item : composedSchema.getAllOf()){
+                        if (item.get$ref() != null){
+                            processRefSchema(item,file);
+                        }
+                    }
+
+                }else if (composedSchema.getOneOf() != null){
```

Comment:
replace `else if` with `if`, as allOf, anyOf and oneOf can theoretically coexist, check with a test case